scalatest/src/main/scala/org/scalatest/enablers/PropCheckerAsserting.scala

```diff
@@ -1600,6 +1625,44 @@ object PropCheckerAsserting extends ExpectationPropCheckerAsserting with FutureP
     else
       ""
 
+  /**
+    * This computes the string to display when a property check fails. It's showing quite a bit, so there's a lot
+    * to it.
+    *
+    * @param failure the actual property check failure, which contains lots of stuff we need to show
+    * @param outerEx the outer exception, generally pointing to the forAll itself
+    * @param prettifier the Prettifier that we will use to improve the error displays
+    * @param argNames the names on the property check arguments, if any
+    * @param labels
+    * @return the detailed error message to show to the user
+    */
+  private[enablers] def failureStr(failure: PropertyCheckResult.Failure, outerEx: StackDepthException, prettifier: Prettifier, argNames: Option[List[String]], labels: List[String]): String = {
+    // ex is the *inner* Exception, where we actually threw. If defined, this is typically the line
+    // that the user really cares about:
+    val PropertyCheckResult.Failure(succeeded, ex, names, argsPassed, initSeed) = failure
+
+    FailureMessages.propertyException(prettifier, UnquotedString(outerEx.getClass.getSimpleName)) +
+      ( outerEx.failedCodeFileNameAndLineNumberString match { case Some(s) => " (" + s + ")"; case None => "" }) + EOL +
+      "  " + FailureMessages.propertyFailed(prettifier, succeeded) + EOL + (
+        ex match {
+          case Some(ex: Throwable) if ex.getMessage != null =>
+            "  " + FailureMessages.thrownExceptionsMessage(prettifier, UnquotedString(ex.getMessage)) + EOL
+          case _ => ""
+        }
+      ) + (
+        ex match {
+          case Some(sd: StackDepth) if sd.failedCodeFileNameAndLineNumberString.isDefined =>
+            "  " + FailureMessages.thrownExceptionsLocation(prettifier, UnquotedString(sd.failedCodeFileNameAndLineNumberString.get)) + EOL
+          case _ => ""
+        }
+      ) +
+      "  " + FailureMessages.occurredOnValues + EOL +
+      prettyArgs(getArgsWithSpecifiedNames(argNames, argsPassed), prettifier) + EOL +
+      "  )" +
+      getLabelDisplay(labels.toSet) + EOL +
+      "  " + FailureMessages.initSeed(prettifier, initSeed),
```

Comment:
Just noticed - the trailing comma here has broken the build, and is a merge-blocker.  (It's legal in 2.12, but not before then.)  Please fix.